sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -560,9 +565,24 @@ class Analyzer(
     def extract(
         expressions: Seq[NamedExpression]): (Seq[NamedExpression], Seq[NamedExpression]) = {
       val (windowExpressions, regularExpressions) = expressions.partition(hasWindowFunction)
-      // Also need to extract all UnresolvedAttribute from windowExpressions.
+
+      // 1. Substitute expression in windowExpressions with alias in regularExpressions if possible
+      // For example, select sum(key) as s1, sum(sum(key)) over (...) from src group by value,
+      // we need substitute the inner sum(key) with s1.
+      val regularExpressionWithAlias = regularExpressions.collect {
+        case a @ Alias(child, name) => (takeOffAlias(child), a)
+      }.toMap
+
+      val newWindowExpressions = windowExpressions.map {
+        _.transform {
+          case expr if regularExpressionWithAlias.contains(expr) =>
+            regularExpressionWithAlias.get(expr).get.toAttribute
+        }.asInstanceOf[NamedExpression]
+      }
```

Comment:
I think instead of having `regularExpressionWithAlias`, a better way is to extract all expressions from an `WindowExpression` and push them to the underlying `Projects` and `Aggregates`. Then, a `WindowExpression` will only have `AttributeReference` inside.